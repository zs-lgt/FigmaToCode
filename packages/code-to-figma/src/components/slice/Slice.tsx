import * as React from 'react';
import { StyleOf, BaseNodeProps, SceneNodeProps, LayoutProps, ExportProps } from '../../types';
import {
    LayoutStyleProperties,
    transformLayoutStyleProperties
} from '../../styleTransformers/transformLayoutStyleProperties';
import { useYogaLayout } from '../../hooks/useYogaLayout';
import { StyleSheet } from '../..';
import { YogaStyleProperties } from '../../yoga/YogaStyleProperties';
import { OnLayoutHandlerProps, useOnLayoutHandler } from '../../hooks/useOnLayoutHandler';

export interface SliceProps extends BaseNodeProps, SceneNodeProps, LayoutProps, ExportProps, OnLayoutHandlerProps {
    style?: StyleOf<LayoutStyleProperties & YogaStyleProperties>;
    children?: undefined;
}

export const Slice: React.FC<SliceProps> = props => {
    const nodeRef = React.useRef();

    const style = { ...StyleSheet.flatten(props.style) };

    const sliceProps = {
        ...transformLayoutStyleProperties(style),
        ...props,
        style
    };
    const yogaProps = useYogaLayout({ nodeRef, ...sliceProps });
    useOnLayoutHandler(yogaProps, props);

    return <slice {...sliceProps} {...yogaProps} innerRef={nodeRef} />;
};
