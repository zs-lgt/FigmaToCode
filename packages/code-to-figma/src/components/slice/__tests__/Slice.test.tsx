import * as React from 'react';
const renderer = require('react-test-renderer');
import { Slice } from '../Slice';
import { createFigma } from 'figma-api-stub';

describe('<Slice />', () => {
    beforeEach(() => {
        // @ts-ignore
        global.figma = createFigma({
            simulateErrors: true
        });
    });

    it('Slice without props', () => {
        const tree = renderer.create(<Slice />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Slice with layout style properties', () => {
        const tree = renderer
            .create(
                <Slice
                    style={{
                        width: 100,
                        height: 200,
                        top: 50,
                        left: 20
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
