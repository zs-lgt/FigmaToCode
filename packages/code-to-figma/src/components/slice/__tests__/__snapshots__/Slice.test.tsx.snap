// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slice /> Slice with layout style properties 1`] = `
<slice
  height={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  style={
    Object {
      "height": 200,
      "left": 20,
      "top": 50,
      "width": 100,
    }
  }
  width={100}
  x={20}
  y={50}
/>
`;

exports[`<Slice /> Slice without props 1`] = `
<slice
  innerRef={
    Object {
      "current": undefined,
    }
  }
  style={Object {}}
/>
`;
